<?xml version="1.0" encoding="utf-16"?>
<TestRun>
  <TestGroup>
    <description>Failed Tests</description>
    <TestCase>
      <Name>Crashing.TheCrash</Name>
      <Result>Test Failed - because it CRASHED!</Result>
      <Source>crashingtests.cpp line 11</Source>
      <Error>Message: !! This is probably the test that crashed !!</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.Simple/1  # GetParam() = (1,!)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 23</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:25
Value of: GetParam().s
  Actual: "!"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.Simple/2  # GetParam() = (-1,)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 23</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:24
Value of: GetParam().i
  Actual: -1
Expected: 1</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits/1  # GetParam() = (1,!)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 28</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:30
Value of: GetParam().s
  Actual: "!"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits/2  # GetParam() = (-1,)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 28</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:29
Value of: GetParam().i
  Actual: -1
Expected: 1</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits2/1  # GetParam() = (1,!)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 33</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:35
Value of: GetParam().s
  Actual: "!"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits2/2  # GetParam() = (-1,)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 33</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:34
Value of: GetParam().i
  Actual: -1
Expected: 1</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits3/1  # GetParam() = (1,!)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 38</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:40
Value of: GetParam().s
  Actual: "!"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits3/2  # GetParam() = (-1,)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 38</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:39
Value of: GetParam().i
  Actual: -1
Expected: 1</Error>
    </TestCase>
    <TestCase>
      <Name>LongRunningTests.Test2</Name>
      <Result>Test Failed -  </Result>
      <Source>longrunningtests.cpp line 11</Source>
      <Error>Message: $(SolutionDir)\tests\longrunningtests.cpp:13
Value of: 2
Expected: 1</Error>
    </TestCase>
    <TestCase>
      <Name>ParameterizedTests.Simple/0  # GetParam() = (0,_)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 23</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:24
Value of: GetParam().i
  Actual: 0
Expected: 1

$(SolutionDir)\tests\parameterizedtests.cpp:25
Value of: GetParam().s
  Actual: "_"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>ParameterizedTests.SimpleTraits/0  # GetParam() = (0,_)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 28</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:29
Value of: GetParam().i
  Actual: 0
Expected: 1

$(SolutionDir)\tests\parameterizedtests.cpp:30
Value of: GetParam().s
  Actual: "_"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>ParameterizedTests.SimpleTraits2/0  # GetParam() = (0,_)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 33</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:34
Value of: GetParam().i
  Actual: 0
Expected: 1

$(SolutionDir)\tests\parameterizedtests.cpp:35
Value of: GetParam().s
  Actual: "_"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>ParameterizedTests.SimpleTraits3/0  # GetParam() = (0,_)</Name>
      <Result>Test Failed -  </Result>
      <Source>parameterizedtests.cpp line 38</Source>
      <Error>Message: $(SolutionDir)\tests\parameterizedtests.cpp:39
Value of: GetParam().i
  Actual: 0
Expected: 1

$(SolutionDir)\tests\parameterizedtests.cpp:40
Value of: GetParam().s
  Actual: "_"
Expected: ""</Error>
    </TestCase>
    <TestCase>
      <Name>TestMath.AddFails</Name>
      <Result>Test Failed -  </Result>
      <Source>basictests.cpp line 29</Source>
      <Error>Message: $(SolutionDir)\tests\basictests.cpp:30
Value of: Add(10, 10)
  Actual: 20
Expected: 1000</Error>
    </TestCase>
    <TestCase>
      <Name>TestMath.Crash</Name>
      <Result>Test Failed -  </Result>
      <Source>basictests.cpp line 39</Source>
      <Error>Message: unknown file
SEH exception with code 0xc0000005 thrown in the test body.</Error>
    </TestCase>
    <TestCase>
      <Name>TheFixture.AddFails</Name>
      <Result>Test Failed -  </Result>
      <Source>fixturetests.cpp line 11</Source>
      <Error>Message: $(SolutionDir)\tests\fixturetests.cpp:12
Value of: Add(10, 10)
  Actual: 20
Expected: 1000</Error>
    </TestCase>
    <TestCase>
      <Name>TheFixture.Crash</Name>
      <Result>Test Failed -  </Result>
      <Source>fixturetests.cpp line 21</Source>
      <Error>Message: unknown file
SEH exception with code 0xc0000005 thrown in the test body.</Error>
    </TestCase>
  </TestGroup>
  <TestGroup>
    <description>Skipped Tests</description>
    <TestCase>
      <Name>Crashing.AddFails</Name>
      <Result>Test Skipped - Crashing.AddFails</Result>
      <Source>crashingtests.cpp line 16</Source>
      <Error>Message: reason is probably a crash of test Crashing.TheCrash</Error>
    </TestCase>
    <TestCase>
      <Name>Crashing.AddPasses</Name>
      <Result>Test Skipped - Crashing.AddPasses</Result>
      <Source>crashingtests.cpp line 21</Source>
      <Error>Message: reason is probably a crash of test Crashing.TheCrash</Error>
    </TestCase>
    <TestCase>
      <Name>Crashing.LongRunning</Name>
      <Result>Test Skipped - Crashing.LongRunning</Result>
      <Source>crashingtests.cpp line 26</Source>
      <Error>Message: reason is probably a crash of test Crashing.TheCrash</Error>
    </TestCase>
  </TestGroup>
  <TestGroup>
    <description>Passed Tests</description>
    <TestCase>
      <Name>CommandArgs.TestDirectoryIsSet</Name>
      <Result>Test Passed -  </Result>
      <Source>basictests.cpp line 23</Source>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.Simple/0  # GetParam() = (1,)</Name>
      <Result>Test Passed -  </Result>
      <Source>parameterizedtests.cpp line 23</Source>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits/0  # GetParam() = (1,)</Name>
      <Result>Test Passed -  </Result>
      <Source>parameterizedtests.cpp line 28</Source>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits2/0  # GetParam() = (1,)</Name>
      <Result>Test Passed -  </Result>
      <Source>parameterizedtests.cpp line 33</Source>
    </TestCase>
    <TestCase>
      <Name>InstantiationName/ParameterizedTests.SimpleTraits3/0  # GetParam() = (1,)</Name>
      <Result>Test Passed -  </Result>
      <Source>parameterizedtests.cpp line 38</Source>
    </TestCase>
    <TestCase>
      <Name>LongRunningTests.Test1</Name>
      <Result>Test Passed -  </Result>
      <Source>longrunningtests.cpp line 5</Source>
    </TestCase>
    <TestCase>
      <Name>TestMath.AddPasses</Name>
      <Result>Test Passed -  </Result>
      <Source>basictests.cpp line 34</Source>
    </TestCase>
    <TestCase>
      <Name>TestMath.AddPassesWithTraits</Name>
      <Result>Test Passed -  </Result>
      <Source>basictests.cpp line 45</Source>
    </TestCase>
    <TestCase>
      <Name>TestMath.AddPassesWithTraits2</Name>
      <Result>Test Passed -  </Result>
      <Source>basictests.cpp line 50</Source>
    </TestCase>
    <TestCase>
      <Name>TestMath.AddPassesWithTraits3</Name>
      <Result>Test Passed -  </Result>
      <Source>basictests.cpp line 55</Source>
    </TestCase>
    <TestCase>
      <Name>TheFixture.AddPasses</Name>
      <Result>Test Passed -  </Result>
      <Source>fixturetests.cpp line 16</Source>
    </TestCase>
    <TestCase>
      <Name>TheFixture.AddPassesWithTraits</Name>
      <Result>Test Passed -  </Result>
      <Source>fixturetests.cpp line 27</Source>
    </TestCase>
    <TestCase>
      <Name>TheFixture.AddPassesWithTraits2</Name>
      <Result>Test Passed -  </Result>
      <Source>fixturetests.cpp line 32</Source>
    </TestCase>
    <TestCase>
      <Name>TheFixture.AddPassesWithTraits3</Name>
      <Result>Test Passed -  </Result>
      <Source>fixturetests.cpp line 37</Source>
    </TestCase>
  </TestGroup>
  <TestOutput>------ Run test started ------
Found 4 tests in executable ${SolutionDir}\Debug\CrashingTests.exe
Found 31 tests in executable ${SolutionDir}\Debug\Tests.exe
Running 35 tests...
Warning: Test teardown batch returned exit code 1, executed command: '${SolutionDir}\Tests\Returns1.bat'
Test execution completed.
========== Run test finished: 35 run (${RunTime}) ==========
</TestOutput>
</TestRun>